diff --git a/src/build.c b/src/build.c
index a5deb54fc6..8d9de5bc72 100644
--- a/src/build.c
+++ b/src/build.c
@@ -2626,10 +2626,15 @@ void sqlite3EndTable(
   ** table itself.  So mark it read-only.
   */
   if( db->init.busy ){
+#ifdef MAGMA_ENABLE_FIXES
     if( pSelect || (!IsOrdinaryTable(p) && db->init.newTnum) ){
       sqlite3ErrorMsg(pParse, "");
       return;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", pSelect != 0);
+#endif
     p->tnum = db->init.newTnum;
     if( p->tnum==1 ) p->tabFlags |= TF_Readonly;
   }
