diff --git a/src/where.c b/src/where.c
--- a/src/where.c
+++ b/src/where.c
@@ -3535,9 +3535,14 @@ static int whereLoopAddBtreeIndex(
    && pProbe->hasStat1!=0
    && OptimizationEnabled(db, SQLITE_SkipScan)
    && pProbe->aiRowLogEst[saved_nEq+1]>=42  /* TUNING: Minimum for skip-scan */
    && pSrc->fg.fromExists==0
+#ifdef MAGMA_ENABLE_FIXES
    && (rc = whereLoopResize(db, pNew, pNew->nLTerm+1))==SQLITE_OK
+#endif
   ){
+#ifdef MAGMA_ENABLE_CANARIES
+      MAGMA_LOG("%MAGMA_BUG%", pNew->nLSlot < (pNew->nLTerm+1));
+#endif
     LogEst nIter;
     pNew->u.btree.nEq++;
     pNew->nSkip++;
