diff --git a/src/select.c b/src/select.c
index cf25c8e678..53cc27c4cc 100644
--- a/src/select.c
+++ b/src/select.c
@@ -6140,10 +6140,17 @@ static int selectExpander(Walker *pWalker, Select *p){
 
   /* Process NATURAL keywords, and ON and USING clauses of joins.
   */
-  assert( db->mallocFailed==0 || pParse->nErr!=0 );
-  if( pParse->nErr || sqlite3ProcessJoin(pParse, p) ){
-    return WRC_Abort;
-  }
+
+  if(
+#ifdef MAGMA_ENABLE_FIXES
+    pParse->nErr ||
+#endif
+    db->mallocFailed || sqlite3ProcessJoin(pParse, p) ){
+      return WRC_Abort;
+    }
+#ifdef MAGMA_ENABLE_CANARIES
+  MAGMA_LOG("%MAGMA_BUG%", pParse->nErr);
+#endif
 
   /* For every "*" that occurs in the column list, insert the names of
   ** all columns in all tables.  And for every TABLE.* insert the names
