diff --git a/src/select.c b/src/select.c
--- a/src/select.c
+++ b/src/select.c
@@ -8293,11 +8293,18 @@ int sqlite3Select(
   if( (p->selFlags & (SF_Distinct|SF_Aggregate))==SF_Distinct
    && sqlite3CopySortOrder(pEList, sSort.pOrderBy)
    && sqlite3ExprListCompare(pEList, sSort.pOrderBy, -1)==0
    && OptimizationEnabled(db, SQLITE_GroupByOrder)
-#ifndef SQLITE_OMIT_WINDOWFUNC
+#ifdef MAGMA_ENABLE_FIXES
+ #ifndef SQLITE_OMIT_WINDOWFUNC
    && p->pWin==0
+ #endif
 #endif
   ){
+#ifdef MAGMA_ENABLE_CANARIES
+#ifndef SQLITE_OMIT_WINDOWFUNC
+    MAGMA_LOG("%MAGMA_BUG%", p->pWin != 0);
+#endif
+#endif
     p->selFlags &= ~(u32)SF_Distinct;
     pGroupBy = p->pGroupBy = sqlite3ExprListDup(db, pEList, 0);
     if( pGroupBy ){
