diff --git a/ssl/t1_lib.c b/ssl/t1_lib.c
index 5a7201b7a2..56f9bdbf8b 100644
--- a/ssl/t1_lib.c
+++ b/ssl/t1_lib.c
@@ -3184,10 +3184,17 @@ SSL_TICKET_STATUS tls_decrypt_ticket(SSL_CONNECTION *s,
     }
 
     /* Sanity check ticket length: must exceed keyname + IV + HMAC */
+#ifdef MAGMA_ENABLE_FIXES
     if (eticklen <= TLSEXT_KEYNAME_LENGTH + ivlen + mlen) {
+#else
+    if (eticklen < 48) {
+#endif
         ret = SSL_TICKET_NO_DECRYPT;
         goto end;
     }
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", eticklen <= TLSEXT_KEYNAME_LENGTH + ivlen + mlen);
+#endif
     eticklen -= mlen;
     /* Check HMAC of encrypted ticket */
     if (ssl_hmac_update(hctx, etick, eticklen) <= 0
