diff --git a/autogen.sh b/autogen.sh
index 2e6c6bd0..32ab8ace 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -4,27 +4,19 @@ retval=0
 
 set -x
 
-case `uname` in
-  Darwin*)
-    glibtoolize --force --copy || retval=$?
-    ;;
-  *)
-    libtoolize --force --copy || retval=$?
-    ;;
-esac
-aclocal -I ./m4 || retval=$?
-autoheader || retval=$?
-automake --foreign --add-missing --copy || retval=$?
-autoconf || retval=$?
+autoreconf --install --force || retval=$?
+
 # Get latest config.guess and config.sub from upstream master since
-# these are often out of date.
+# these are often out of date.  This requires network connectivity and
+# sometimes the site is down, a failure here does not result in
+# failure of the whole script.
 for file in config.guess config.sub
 do
     echo "$0: getting $file..."
-    wget -q --timeout=5 -O config/$file.tmp \
+    wget --timeout=5 -O config/$file.tmp \
       "https://git.savannah.gnu.org/cgit/config.git/plain/${file}" \
       && mv -f config/$file.tmp config/$file \
-      && chmod a+x config/$file || retval=$?
+      && chmod a+x config/$file
     rm -f config/$file.tmp
 done
 
