diff --git a/libtiff/tif_lzw.c b/libtiff/tif_lzw.c
index 05ac8aa1..b9b0e9b9 100644
--- a/libtiff/tif_lzw.c
+++ b/libtiff/tif_lzw.c
@@ -973,9 +973,16 @@ static int LZWDecodeCompat(TIFF *tif, uint8_t *op0, tmsize_t occ0, uint16_t s)
             tp = op + len;
             do
             {
+#ifdef MAGMA_ENABLE_CANARIES
+				MAGMA_LOG("%MAGMA_BUG%", tp <= op); // or use == instead of <= so the bug is reported once only
+#endif
                 *--tp = codep->value;
                 codep = codep->next;
+#ifdef MAGMA_ENABLE_FIXES
             } while (codep && tp > op);
+#else
+			} while (codep);
+#endif
             assert(occ >= len);
             op += len;
             occ -= len;
