diff --git a/libtiff/tif_ojpeg.c b/libtiff/tif_ojpeg.c
index 57adcff8..513753f0 100644
--- a/libtiff/tif_ojpeg.c
+++ b/libtiff/tif_ojpeg.c
@@ -862,12 +862,14 @@ static int OJPEGDecode(TIFF *tif, uint8_t *buf, tmsize_t cc, uint16_t s)
     static const char module[] = "OJPEGDecode";
     OJPEGState *sp = (OJPEGState *)tif->tif_data;
     (void)s;
+#ifdef MAGMA_ENABLE_FIXES
     if (!sp->decoder_ok)
     {
         TIFFErrorExtR(tif, module,
                       "Cannot decode: decoder not correctly initialized");
         return 0;
     }
+#endif
     if (sp->libjpeg_session_active == 0)
     {
         /* This should normally not happen, except that it does when */
@@ -881,10 +883,15 @@ static int OJPEGDecode(TIFF *tif, uint8_t *buf, tmsize_t cc, uint16_t s)
                       "Cannot decode: libjpeg_session_active == 0");
         return 0;
     }
+#ifdef MAGMA_ENABLE_FIXES
     if (sp->error_in_raw_data_decoding)
     {
         return 0;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", sp->decoder_ok == 0);
+#endif
     if (sp->libjpeg_jpeg_query_style == 0)
     {
         if (OJPEGDecodeRaw(tif, buf, cc) == 0)
