diff --git a/libtiff/tif_ojpeg.c b/libtiff/tif_ojpeg.c
index f94d2a4e..3898558d 100644
--- a/libtiff/tif_ojpeg.c
+++ b/libtiff/tif_ojpeg.c
@@ -841,6 +841,7 @@ static int OJPEGDecode(TIFF *tif, uint8_t *buf, tmsize_t cc, uint16_t s)
     static const char module[] = "OJPEGDecode";
     OJPEGState *sp = (OJPEGState *)tif->tif_data;
     (void)s;
+#ifdef MAGMA_ENABLE_FIXES
     if (!sp->decoder_ok)
     {
         memset(buf, 0, (size_t)cc);
@@ -848,6 +849,7 @@ static int OJPEGDecode(TIFF *tif, uint8_t *buf, tmsize_t cc, uint16_t s)
                       "Cannot decode: decoder not correctly initialized");
         return 0;
     }
+#endif
     if (sp->libjpeg_session_active == 0)
     {
         memset(buf, 0, (size_t)cc);
@@ -862,11 +864,16 @@ static int OJPEGDecode(TIFF *tif, uint8_t *buf, tmsize_t cc, uint16_t s)
                       "Cannot decode: libjpeg_session_active == 0");
         return 0;
     }
+#ifdef MAGMA_ENABLE_FIXES
     if (sp->error_in_raw_data_decoding)
     {
         memset(buf, 0, (size_t)cc);
         return 0;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", sp->decoder_ok == 0);
+#endif
     if (sp->libjpeg_jpeg_query_style == 0)
     {
         if (OJPEGDecodeRaw(tif, buf, cc) == 0)
