diff --git a/libtiff/tif_dir.c b/libtiff/tif_dir.c
index 47ad21e0..06689f24 100644
--- a/libtiff/tif_dir.c
+++ b/libtiff/tif_dir.c
@@ -164,7 +164,7 @@ static int setExtraSamples(TIFF *tif, va_list ap, uint32_t *v)
                 return 0;
         }
     }
-
+#ifdef MAGMA_ENABLE_FIXES
     if (td->td_transferfunction[0] != NULL &&
         (td->td_samplesperpixel - *v > 1) &&
         !(td->td_samplesperpixel - td->td_extrasamples > 1))
@@ -177,6 +177,11 @@ static int setExtraSamples(TIFF *tif, va_list ap, uint32_t *v)
         _TIFFfreeExt(tif, td->td_transferfunction[0]);
         td->td_transferfunction[0] = NULL;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+        MAGMA_LOG("%MAGMA_BUG%", MAGMA_AND(td->td_transferfunction[0] != NULL, MAGMA_AND(td->td_samplesperpixel - *v > 1, \
+            !(td->td_samplesperpixel - td->td_extrasamples > 1))));
+#endif
 
     td->td_extrasamples = (uint16_t)*v;
     _TIFFsetShortArrayExt(tif, &td->td_sampleinfo, va, td->td_extrasamples);
@@ -324,6 +329,7 @@ static int _TIFFVSetField(TIFF *tif, uint32_t tag, va_list ap)
                 goto badvalue;
             if (v != td->td_samplesperpixel)
             {
+#ifdef MAGMA_ENABLE_FIXES
                 /* See http://bugzilla.maptools.org/show_bug.cgi?id=2500 */
                 if (td->td_sminsamplevalue != NULL)
                 {
@@ -359,6 +365,13 @@ static int _TIFFVSetField(TIFF *tif, uint32_t tag, va_list ap)
                     TIFFClrFieldBit(tif, FIELD_TRANSFERFUNCTION);
                     _TIFFfreeExt(tif, td->td_transferfunction[0]);
                     td->td_transferfunction[0] = NULL;
                 }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+            MAGMA_LOG("%MAGMA_BUG%", MAGMA_OR(td->td_sminsamplevalue != NULL, \
+                MAGMA_OR(td->td_smaxsamplevalue != NULL, \
+                MAGMA_AND(td->td_transferfunction[0] != NULL, MAGMA_AND(v - td->td_extrasamples > 1, \
+                   !(td->td_samplesperpixel - td->td_extrasamples > 1))))));
+#endif
             }
             td->td_samplesperpixel = (uint16_t)v;
