diff --git a/poppler/JPEG2000Stream.cc b/poppler/JPEG2000Stream.cc
index 67d28a8c..a678d3d8 100644
--- a/poppler/JPEG2000Stream.cc
+++ b/poppler/JPEG2000Stream.cc
@@ -242,9 +242,16 @@ void JPXStream::init()
     }
 
     int bufSize = BUFFER_INITIAL_SIZE;
+#ifdef MAGMA_ENABLE_FIXES
     if (oLen.isInt() && oLen.getInt() > 0) {
         bufSize = oLen.getInt();
     }
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", oLen.getInt() < 0);
+#endif
+    if (oLen.isInt()) bufSize = oLen.getInt();
+#endif
 
     bool indexed = false;
     if (cspace.isArray() && cspace.arrayGetLength() > 0) {
