diff --git a/splash/SplashXPathScanner.cc b/splash/SplashXPathScanner.cc
index 3d7db3ac..e7390903 100644
--- a/splash/SplashXPathScanner.cc
+++ b/splash/SplashXPathScanner.cc
@@ -449,9 +449,14 @@ void SplashXPathScanner::clipAALine(SplashBitmap *aaBuf, const int *x0, const in
         xx = *x0 * splashAASize;
         if (yy >= yyMin && yy <= yyMax) {
             const int intersectionIndex = splashAASize * y + yy - yMin;
+#ifdef MAGMA_ENABLE_FIXES
             if (unlikely(intersectionIndex < 0 || (unsigned)intersectionIndex >= allIntersections.size())) {
                 break;
             }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+            MAGMA_LOG("%MAGMA_BUG%", MAGMA_OR(intersectionIndex < 0, (unsigned)intersectionIndex >= allIntersections.size()));
+#endif
             const auto &line = allIntersections[intersectionIndex];
             interIdx = 0;
             interCount = 0;
