diff --git a/poppler/FileSpec.cc b/poppler/FileSpec.cc
index 62782603..985483f8 100644
--- a/poppler/FileSpec.cc
+++ b/poppler/FileSpec.cc
@@ -104,9 +104,14 @@ bool EmbFile::save2(FILE *f)
 {
     int c;
 
+#ifdef MAGMA_ENABLE_FIXES
     if (unlikely(!m_objStr.isStream())) {
         return false;
     }
+#endif
+#ifdef MAGMA_ENABLE_CANARIES
+    MAGMA_LOG("%MAGMA_BUG%", !m_objStr.isStream());
+#endif
 
     m_objStr.streamReset();
     while ((c = m_objStr.streamGetChar()) != EOF) {
