diff --git a/fofi/FoFiTrueType.cc b/fofi/FoFiTrueType.cc
index c6cb8233..59647e4a 100644
--- a/fofi/FoFiTrueType.cc
+++ b/fofi/FoFiTrueType.cc
@@ -1373,15 +1373,27 @@ void FoFiTrueType::cvtSfnts(FoFiOutputFunc outputFunc, void *outputStream, const
                 if ((j = seekTable(t42Tables[i].tag)) >= 0 && checkRegion(tables[j].offset, tables[j].len)) {
                     dumpString(std::span(file + tables[j].offset, tables[j].len), outputFunc, outputStream);
                 } else if (needVerticalMetrics && i == t42VheaTable) {
+#ifdef MAGMA_ENABLE_FIXES
                     if (unlikely(length > (int)sizeof(vheaTab))) {
                         error(errSyntaxWarning, -1, "length bigger than vheaTab size");
                         length = sizeof(vheaTab);
                     }
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+                    MAGMA_LOG("%MAGMA_BUG%", length > (int)sizeof(vheaTab));
+#endif
+#endif
                     dumpString(vheaTab, outputFunc, outputStream);
                 } else if (needVerticalMetrics && i == t42VmtxTable) {
+#ifdef MAGMA_ENABLE_FIXES
                     if (unlikely(length > (int)vmtxTab.size())) {
                         error(errSyntaxWarning, -1, "length bigger than vmtxTab size");
                     }
+#else
+#ifdef MAGMA_ENABLE_CANARIES
+                    MAGMA_LOG("%MAGMA_BUG%", length > (int)vmtxTab.size());
+#endif
+#endif
                     dumpString(vmtxTab, outputFunc, outputStream);
                 }
             }
